lib/Client.js

```diff
@@ -1588,6 +1722,9 @@ class Client extends EventEmitter {
         if(this.channelGuildMap[channelID] && this.guilds.get(this.channelGuildMap[channelID])) {
             return this.guilds.get(this.channelGuildMap[channelID]).channels.get(channelID);
         }
+        if(this.threadGuildMap[channelID] && this.guilds.get(this.threadGuildMap[channelID])) {
+            return this.guilds.get(this.threadGuildMap[channelID]).channels.get(channelID);
```

Comment:
Threads are not included in `<Guild>.channels`, this needs to be `<Guild>.threads.get(channelID);`